src/mpi/init/mpi_init.h

```diff
@@ -75,7 +75,8 @@ static inline void MPII_post_init_memory_tracing(void)
 {
 #ifdef USE_MEMORY_TRACING
 #ifdef MPICH_IS_THREADED
-    MPL_trconfig(MPIR_Process.comm_world->rank, MPIR_ThreadInfo.isThreaded);
+    MPL_trconfig(MPIR_Process.comm_world->rank,
+                 MPIR_ThreadInfo.thread_provided == MPI_THREAD_MULTIPLE);
```

Comment:
Remove preprocessor guards here, too?

Reply:
Updated in commit https://github.com/pmodels/mpich/pull/4199/commits/96594d579ca73407a6d63443266acc731e1bc8b5